src/main/java/net/mcreator/element/types/Dimension.java

```diff
@@ -105,21 +106,25 @@ public Dimension(ModElement element) {
 	}
 
 	@Override public Collection<BaseType> getBaseTypesProvided() {
+		List<BaseType> baseTypes = new ArrayList<>();
 		if (enablePortal)
-			return List.of(BaseType.BLOCK, BaseType.ITEM);
-		else
-			return Collections.emptyList();
+			baseTypes.add(BaseType.BLOCK);
+		if (enableIgniter)

```

Comment:
If igniter is enabled, but portal is disabled, igniter should not be generated either, so when checking for igniter, you should probably && with portal

Reply:
Igniter can't be enabled if portal is disabled, that doesn't make sense

Reply:
I can 100% agee, however this state can easily be achieved based on current implementation:

![image](https://github.com/MCreator/MCreator/assets/16374228/81301686-57eb-4f9c-83a5-84dac9ef60ba)

To fix this properly, I would recommend making method hasIgniterItem() that checks for igniter and portal, and replace enableIgniter in all places with this method in templates and generator files